Check for carbon in elements for Hill

CasperWA opened the pull request:
Fixes #585.

If carbon is not present, the elements should be sorted alphabetically (including hydrogen).

This changes a test to the corrected definition of Hill notation.

ml-evs reviewed (approved):
LGTM, thanks for spotting this

ml-evs commented:
Actually, why don't the tests I just added fail, e.g.

https://github.com/Materials-Consortia/optimade-python-tools/blob/d48b7a80e24aad591857c386498774037bbec89c/tests/models/test_structures.py#L83-L85

ml-evs commented:
> Actually, why don't the tests I just added fail, e.g.
> 
> https://github.com/Materials-Consortia/optimade-python-tools/blob/d48b7a80e24aad591857c386498774037bbec89c/tests/models/test_structures.py#L83-L85

Okay mb, I see that test was updated (it just looked different to my local version)

CasperWA commented:
> Actually, why don't the tests I just added fail, e.g.
> 
> https://github.com/Materials-Consortia/optimade-python-tools/blob/d48b7a80e24aad591857c386498774037bbec89c/tests/models/test_structures.py#L83-L85

Because I updated this exact test (as mentioned in the OP and found in the `git diff`).
